engine/src/main/java/org/terasology/entitySystem/entity/internal/PojoEntityManager.java

```diff
@@ -467,7 +407,8 @@ public boolean isActiveEntity(long id) {
     @Override
     //Todo: implement iterating over multiple pools
     public Iterable<Component> iterateComponents(long entityId) {
-        return globalPool.getComponentStore().iterateComponents(entityId);
+        return Iterables.concat(globalPool.getComponentStore().iterateComponents(entityId),
+                sectorManager.getComponentStore().iterateComponents(entityId));
```

Comment:
Ah, this was just the general form I was using to combine iterators. It works because the entity is only in one pool at a time, but it would be better to do it by just finding out which pool the entity is in and iterating over it in that pool. That would be more efficient, communicate better what's actually happening, and probably be more resistant to bugs.

There's currently no way of an entity ending up in multiple pools (and I don't think there should be), so I don't think this would actually be a problem in terms of program correctness, but it's certainly not the best way to do it.